src/UsgsAstroFramePlugin.cpp

```diff
@@ -293,191 +289,139 @@ csm::Model *UsgsAstroFramePlugin::constructModelFromISD(const csm::Isd &imageSup
-  }
+  sensorModel->m_currentParameterValue[3] = atof(imageSupportData.param("sensor_orientation", 0).c_str());
+  sensorModel->m_currentParameterValue[4] = atof(imageSupportData.param("sensor_orientation", 1).c_str());
+  sensorModel->m_currentParameterValue[5] = atof(imageSupportData.param("sensor_orientation", 2).c_str());
+  sensorModel->m_currentParameterValue[6] = atof(imageSupportData.param("sensor_orientation", 3).c_str());
 
-  sensorModel->m_ephemerisTime = atof(imageSupportData.param("ephemeris_time").c_str());
-  if (imageSupportData.param("ephemeris_time") == "") {
-    missingKeywords.push_back("ephemeris_time");
+  if (imageSupportData.param("sensor_orientation", 0) == "") {
+    missingKeywords.push_back("sensor_orientation");
   }
 
-  sensorModel->m_originalHalfSamples =
-      atof(imageSupportData.param("original_half_samples").c_str());
+  sensorModel->m_odtX[0] = atof(imageSupportData.param("optical_distortion_x", 0).c_str());
+  sensorModel->m_odtX[1] = atof(imageSupportData.param("optical_distortion_x", 1).c_str());
+  sensorModel->m_odtX[2] = atof(imageSupportData.param("optical_distortion_x", 2).c_str());
+  sensorModel->m_odtX[3] = atof(imageSupportData.param("optical_distortion_x", 3).c_str());
+  sensorModel->m_odtX[4] = atof(imageSupportData.param("optical_distortion_x", 4).c_str());
+  sensorModel->m_odtX[5] = atof(imageSupportData.param("optical_distortion_x", 5).c_str());
+  sensorModel->m_odtX[6] = atof(imageSupportData.param("optical_distortion_x", 6).c_str());
+  sensorModel->m_odtX[7] = atof(imageSupportData.param("optical_distortion_x", 7).c_str());
+  sensorModel->m_odtX[8] = atof(imageSupportData.param("optical_distortion_x", 8).c_str());
+  sensorModel->m_odtX[9] = atof(imageSupportData.param("optical_distortion_x", 9).c_str());
 
-  sensorModel->m_boresight[0] = atof(imageSupportData.param("boresight", 0).c_str());
-  sensorModel->m_boresight[1] = atof(imageSupportData.param("boresight", 1).c_str());
-  sensorModel->m_boresight[2] = atof(imageSupportData.param("boresight", 2).c_str());
+  sensorModel->m_odtY[0] = atof(imageSupportData.param("optical_distortion_y", 0).c_str());
+  sensorModel->m_odtY[1] = atof(imageSupportData.param("optical_distortion_y", 1).c_str());
+  sensorModel->m_odtY[2] = atof(imageSupportData.param("optical_distortion_y", 2).c_str());
+  sensorModel->m_odtY[3] = atof(imageSupportData.param("optical_distortion_y", 3).c_str());
+  sensorModel->m_odtY[4] = atof(imageSupportData.param("optical_distortion_y", 4).c_str());
+  sensorModel->m_odtY[5] = atof(imageSupportData.param("optical_distortion_y", 5).c_str());
+  sensorModel->m_odtY[6] = atof(imageSupportData.param("optical_distortion_y", 6).c_str());
+  sensorModel->m_odtY[7] = atof(imageSupportData.param("optical_distortion_y", 7).c_str());
+  sensorModel->m_odtY[8] = atof(imageSupportData.param("optical_distortion_y", 8).c_str());
+  sensorModel->m_odtY[9] = atof(imageSupportData.param("optical_distortion_y", 9).c_str());
```

Comment:
not entirely sure what you are asking. I don't know very much about the distortion models, so you need to be more specific. like what should be the expectation for arrays whose length  is < 10. pad with zeroes? because that's the current behavior. If not, what else?

Reply:
Disclaimer: This could be the wrong way to do this.
I coded the state stuff to utilize vectors. I wonder if the distortion could do that? If we assume all of the coefficients are floats, I think we can just use the standard type? This also plays nice with the JSON lib.

https://github.com/USGS-Astrogeology/CSM-CameraModel/blob/master/src/UsgsAstroFramePlugin.cpp#L266